packages/babel-helpers/src/helpers.js

```diff
@@ -472,7 +472,15 @@ helpers.construct = helper("7.0.0-beta.0")`
     try {
       // If the internal slots aren't set, this throws an error similar to
       //   TypeError: this is not a Date object.
+      // Date#toString is generic in ES2015 [1] and it doesn't throw, so we also
+      // check Map#get (which on the other hand doesn't exist in ES5 browsers).
+      //
+      // [1]: https://github.com/tc39/ecma262/issues/1268#issuecomment-410104832
+
       Date.prototype.toString.call(Reflect.construct(Date, [], function() {}));
```

Comment:
Per [20.3.4.1](https://www.ecma-international.org/ecma-262/6.0/#sec-date.prototype.tostring), if `Reflect.construct(Date, [], function() {})` is not a Date object, it must returns `"Invalid Date"`, thus it suffices to check if it does _not_ return "Invalid Date".

It helps shrink the code size a bit as it is included in the `wrapNativeClass` helpers.